static/app/utils/useLocation.tsx

```diff
@@ -0,0 +1,10 @@
+import {useContext} from 'react';
+
+import {RouteContext} from 'sentry/views/routeContext';
+
+function useLocation() {
```

Comment:
given 
```
  interface WithRouterProps<P = Record<string, string>, Q = any> {
    location: Location<Q>;
    params: P;
    router: InjectedRouter<P, Q>;
    routes: PlainRoute[];
  }
  ```
we should allow useLocation to accept a type argument so that we can pass that along and not lock ourselves of the type :)
```
// node_modules/@types/history/index.d.ts
type DefaultQuery<T = string> = {
    [key: string]: T | T[] | null | undefined;
};
```

This will allow usage like
```ts
const location = useLocation<{orgId?: string}>();
location.query.orgId // is of type string
```

Reply:
I need to manually verify if that's _exactly_ how it works, but we should nevertheless have the option to type arguments to WithRouterProps

Reply:
@JonasBa yea Im not too familiar with this. were you able to verify this?

Reply:
Just checked - this is the change that I would propose here
```tsx
import {useContext} from 'react';
import {Location, Query} from 'history';

import {RouteContext} from 'sentry/views/routeContext';

export function useLocation<Q extends Query>(): Location<Q> {
  const route = useContext(RouteContext);
  if (route === null) {
    throw new Error('useLocation called outside of routes provider');
  }
  return route?.location;
}

const Test = () => {
  const query = useLocation<{customQuery: string[]}>();
  //   ^^ query is of type {customQuery: string[]}
};

```